test_othermlpackages/lightgbm/test_lightgbm.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import pytest
+
+import test_othermlpackages.package_test_common as ptc
+
+from fairlearn.reductions import DemographicParity
+
+try:
+    import lightgbm as lgb
+except ImportError:
+    pytestmark = pytest.mark.skip(reason="LightGBM not installed")
```

Comment:
Won't these be executed deliberately? We may just accidentally skip them and not notice with this code.

Reply:
I'm not sure what you mean. This lets you run `python -m pytest test_othermlpackages` and it will automatically skip any tests which don't have their dependencies installed.

Reply:
... which means it'll pass even if something isn't set up right and the installation doesn't have the right packages (for whatever reason). What we really want is to deliberately run it, and then it should never skip, right?

Reply:
Point taken. This was my workaround to allow sharing of the common code while putting tests in their own directory. But yes, as you say, if the installation is bad on the test machine, then we will have spuriously green test runs.

Reply:
Agreed! I've used this workaround plenty of times, but in this case it may be better to fail explicitly :-)

Reply:
Directory structure flattened, and this `try` block removed.